tests/kernel/threads/lifecycle/lifecycle_api/src/main.c

```diff
@@ -25,21 +25,27 @@ extern void test_threads_cancel_started(void);
 extern void test_threads_abort_self(void);
 extern void test_threads_abort_others(void);
 
+__kernel struct k_thread tdata;
+#define STACK_SIZE (256 + CONFIG_TEST_EXTRA_STACKSIZE)
+K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+
 /*test case main entry*/
 void test_main(void)
 {
+	k_thread_access_grant(k_current_get(), &tdata, tstack, NULL);
+
 	ztest_test_suite(test_threads_lifecycle,
-			 ztest_unit_test(test_threads_spawn_params),
+			 ztest_user_unit_test(test_threads_spawn_params),
 			 ztest_unit_test(test_threads_spawn_priority),
-			 ztest_unit_test(test_threads_spawn_delay),
+			 ztest_user_unit_test(test_threads_spawn_delay),
 			 ztest_unit_test(test_threads_spawn_forever),
 			 ztest_unit_test(test_threads_suspend_resume_cooperative),
```

Comment:
just out of curiosity, what will be the impact if we run the spawn and suspend tests in user mode? i see that suspend/resume just passing the priority number it wants, how does this impact the user mode execution.?

Reply:
they will fail, k_thread_priority_set() only lets threads downgrade priority, not upgrade it